sdks/java/core/src/main/java/org/apache/beam/sdk/util/FileIOChannelFactory.java

```diff
@@ -159,4 +162,50 @@ public String resolve(String path, String other) throws IOException {
   public Path toPath(String path) {
     return specToFile(path).toPath();
   }
+
+  @Override
+  public void copy(List<String> srcFilenames, List<String> destFilenames) throws IOException {
+    checkArgument(
+        srcFilenames.size() == destFilenames.size(),
+        "Number of source files %s must equal number of destination files %s",
+        srcFilenames.size(),
+        destFilenames.size());
+    int numFiles = srcFilenames.size();
+    for (int i = 0; i < numFiles; i++) {
+      String src = srcFilenames.get(i);
+      String dst = destFilenames.get(i);
+      LOG.debug("Copying {} to {}", src, dst);
+      copyOne(src, dst);
+    }
+  }
+
+  private void copyOne(String source, String destination) throws IOException {
+    try {
+      // Copy the source file, replacing the existing destination.
+      // Paths.get(x) will not work on win cause of the ":" after the drive letter
+      Files.copy(
+          new File(source).toPath(),
+          new File(destination).toPath(),
+          StandardCopyOption.REPLACE_EXISTING);
+    } catch (NoSuchFileException e) {
+      LOG.debug("{} does not exist.", source);
+      // Suppress exception if file does not exist.
```

Comment:
I agree.
However, improving the IOChannelFactory interface is out of the scope of the PR.